drivers/mpu9250/MPU9250.cpp

```diff
@@ -47,36 +48,107 @@
 
 using namespace DriverFramework;
 
+/* TODO: put this in a nice package, make sure everything is inline */
+long int timespec_diff(struct timespec *start, struct timespec *stop);
```

Comment:
I would vote to do everything in microseconds (us). We already have a bunch of helper functions in https://github.com/PX4/DriverFramework/blob/master/framework/src/Time.cpp.